xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-test/xwiki-platform-icon-test-docker/src/test/it/org/xwiki/icon/test/ui/IconThemesRestIT.java

```diff
@@ -0,0 +1,170 @@
+import javax.xml.bind.JAXBContext;
+
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.xwiki.icon.rest.IconThemesResource;
+import org.xwiki.icon.rest.model.jaxb.Icon;
+import org.xwiki.icon.rest.model.jaxb.Icons;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.docker.junit5.UITest;
+import org.xwiki.test.ui.TestUtils;
+
+import static java.util.Collections.singletonList;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+/**
+ * Functional tests of the icon REST endpoint.
+ *
+ * @version $Id$
+ * @since 13.4RC1
+ */
+@UITest
+class IconThemesRestIT
+{
+    @BeforeEach
+    void setUp(TestUtils testUtils)
+    {
+        // Login as superadmin to define the preferences. 
+        testUtils.loginAsSuperAdmin();
+        DocumentReference documentReference = new DocumentReference("xwiki", "XWiki", "XWikiPreferences");
+        testUtils.updateObject(documentReference, "XWiki.XWikiPreferences", 0, "iconTheme", "IconThemes.FontAwesome");
+        // Lougout because the rest of the test must pass as a guest user.
+        testUtils.forceGuestUser();
+    }
+
+    @Test
+    @Order(1)
+    void iconsNoParamKeys(TestUtils testUtils) throws Exception
+    {
+        URI rootIconThemeURI = testUtils.rest().createUri(IconThemesResource.class, new HashMap<>(), "xwiki");
+        GetMethod getMethod =
+            testUtils.rest().executeGet(UriBuilder.fromUri(rootIconThemeURI).segment("icons").build());
+        String body = getMethod.getResponseBodyAsString();
+        assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>"
+            + "<icons xmlns=\"http://www.xwiki.org/icon\"/>", body);
+    }
+
+    @Test
+    @Order(2)
+    void iconsUnknownWiki(TestUtils testUtils) throws Exception
+    {
+        URI rootIconThemeURI = testUtils.rest().createUri(IconThemesResource.class, new HashMap<>(), "nowiki");
+        GetMethod getMethod =
+            testUtils.rest().executeGet(UriBuilder.fromUri(rootIconThemeURI).segment("icons").build());
+        int statusCode = getMethod.getStatusCode();
+        assertEquals(NOT_FOUND.getStatusCode(), statusCode);
+    }
+
+    @Test
+    @Order(3)
+    void icons(TestUtils testUtils) throws Exception
+    {
+        URI rootIconThemeURI = testUtils.rest().createUri(IconThemesResource.class, new HashMap<>(), "xwiki");
+
+        URI uri = UriBuilder.fromUri(rootIconThemeURI)
+            .segment("icons")
+            .queryParam("name", "add", "unknown", "arrow_undo")
+            .build();
+        GetMethod getMethod = testUtils.rest().executeGet(uri);
+        String body = getMethod.getResponseBodyAsString();
+        assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>"
```

Comment:
Note that you could also check the JSON output, which is less verbose.

Reply:
I agree, but in this case we'd need to add a new `executeGetJson`  in `RestTestUtils`, or add a way to configure the `Accept` header on `executeGet` (or a generic way to configure the headers in `RestTestUtils`).

WDYT?

Reply:
You can also force JSON output from the query string, with ``media=json``, but anyway, this is not important.